test/helpers/types/optional_int.go

```diff
@@ -0,0 +1,75 @@
+package types
+
+import (
+	"fmt"
+	"github.com/ViaQ/logerr/log"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+//OptionalInt allows passing an arbitrary int as well matching conditional values
+//for message comparison (e.g. >=6)
+type OptionalInt string
+
+func (oi *OptionalInt) MarshalJSON() ([]byte, error) {
+	return []byte(*oi), nil
+}
+
+func (oi *OptionalInt) UnmarshalJSON(data []byte) error {
+	*oi = OptionalInt(string(data))
+	return nil
+}
+
+func (oi OptionalInt) getParts() (string, int) {
+	comparison := "="
+	value := 0
+	optionalIntRE := regexp.MustCompile(`(?P<comparison>[><=]{1,2})?(?P<value>\d*)`)
```

Comment:
should this be in the package scope?

Reply:
I purposely made it private as i didn't see an immediate need for it to be utilized publicly

Reply:
I meant lifetime, not export status:
```var optionalIntRE = regexp.MustCompile(`(?P<comparison>[><=]{1,2})?(?P<value>\d*)`)

func (oi OptionalInt) getParts() (string, int) {

...
```

Reply:
I left it as-is to ensure the group names stayed close to where they were used.  Those could also be factored into private constants.  I'm indifferent here because it's a test type not being used in a production scenario; efficiency isn't of main concern.  I defer to you how firm your belief is that it should be elsewhere for the change, otherwise I'm find as-is

Reply:
@syedriko I think we will need follow-up PRs anyway to verify we are getting the sequence number.  I'll try to clean up in subsequent PRs